hathor/builder/builder.py

```diff
@@ -239,30 +245,82 @@ def _get_or_create_pubsub(self) -> PubSubManager:
 
     def _get_or_create_rocksdb_storage(self) -> RocksDBStorage:
         assert self._rocksdb_path is not None
-        if self._rocksdb_storage is None:
-            self._rocksdb_storage = RocksDBStorage(path=self._rocksdb_path)
+
+        if self._rocksdb_storage is not None:
+            return self._rocksdb_storage
+
+        kwargs = {}
+        if self._rocksdb_cache_capacity is not None:
+            kwargs = dict(cache_capacity=self._rocksdb_cache_capacity)
+
+        self._rocksdb_storage = RocksDBStorage(
+            path=self._rocksdb_path,
+            **kwargs
+        )
+
         return self._rocksdb_storage
 
     def _get_or_create_tx_storage(self) -> TransactionStorage:
         if self._tx_storage is not None:
             return self._tx_storage
+
         if self._storage_type == StorageType.MEMORY:
             return TransactionMemoryStorage()
-        elif self._storage_type == StorageType.ROCKSDB:
+
+        if self._storage_type == StorageType.ROCKSDB:
             rocksdb_storage = self._get_or_create_rocksdb_storage()
             use_memory_index = self._force_memory_index
-            return TransactionRocksDBStorage(rocksdb_storage, use_memory_indexes=use_memory_index)
+
+            kwargs = {}
+            if self._with_rocksdb_index is not None:
+                kwargs = dict(with_index=self._with_rocksdb_index)
+
+            return TransactionRocksDBStorage(
+                rocksdb_storage,
+                use_memory_indexes=use_memory_index,
+                **kwargs
+            )
+
         raise NotImplementedError
 
+    def _get_event_storage(self) -> EventStorage:
+        if self._storage_type == StorageType.MEMORY:
+            return EventMemoryStorage()
+
+        if self._storage_type == StorageType.ROCKSDB:
+            rocksdb_storage = self._get_or_create_rocksdb_storage()
+            return EventRocksDBStorage(rocksdb_storage)
+
+        raise NotImplementedError
+
+    def with_event_manager(self, *, event_ws_factory: EventWebsocketFactory) -> 'Builder':
```

Comment:
Why `with_` instead of `set_`? From the code, we might rename it to `enable_event_manager(...)`. What do you think?

Reply:
I didn't use `set_` because this method doesn't receive an `EventManager`, it receives an `EventWebsocketFactory` that is used (and required) to build the `EventManager` later.

I agree and changed it to `enable_event_manager()` in https://github.com/HathorNetwork/hathor-core/pull/568/commits/c10b4428a83ee376a7b68ccae29aeabf8a99137e

Reply:
Great! Thanks!